subgraphs/morpho/src/mapping/morpho-compound/comptroller.ts

```diff
@@ -24,17 +24,14 @@ export function handleNewBorrowCap(event: NewBorrowCap): void {
 }
 
 export function handleNewCloseFactor(event: NewCloseFactor): void {
+  const protocol = getOrInitMarketList(MORPHO_COMPOUND_ADDRESS);
   const closeFactor = event.params.newCloseFactorMantissa
     .toBigDecimal()
     .div(exponentToBigDecimal(DEFAULT_DECIMALS))
     .minus(BIGDECIMAL_ONE);
-  const marketList = _MarketList.load(MORPHO_COMPOUND_ADDRESS);
-  if (!marketList) {
-    log.error("[handleNewCloseFactor] Market list not found", []);
-    return;
-  }
-  for (let i = 0; i < marketList.markets.length; i++) {
-    const market = getMarket(marketList.markets[i]);
+
+  for (let i = 0; i < protocol.markets.length; i++) {
```

Comment:
Does it work to use `protocol.markets[]`? I had changed this in the PR I made bc it wasn't building...

That is kinda strange, it looks like the gh action builds just fine..

Reply:
yeah, It is working fine for me.